File: SIEM/app/models/playbook.py
# app/models/playbook.py
# -------------------------------
# Modèle Playbook SOAR — stocké dans l'index ES "playbooks"
#
# Ce que tu dois mettre ici :
#
#   from pydantic import BaseModel, Field
#   from typing import Optional, List
#   from datetime import datetime
#   from enum import Enum
#
#   class PlaybookTrigger(str, Enum):
#       MANUAL = "manual"
#       ALERT_CREATED = "alert_created"
#       ALERT_ESCALATED = "alert_escalated"
#       SCHEDULED = "scheduled"
#       WEBHOOK = "webhook"
#
#   class Playbook(BaseModel):
#       """Playbook d'automatisation SOAR."""
#       id: Optional[str] = None
#       name: str
#       description: Optional[str] = None
#       trigger: PlaybookTrigger = PlaybookTrigger.MANUAL
#       enabled: bool = True
#       steps: List[dict] = []               # [{"action": "enrich_ip", "params": {"ip": "{{source_ip}}", "provider": "virustotal"}}, ...]
#       variables: dict = {}
#       timeout_seconds: int = 300
#       max_retries: int = 3
#       last_executed_at: Optional[datetime] = None
#       execution_count: int = 0
#       created_by: Optional[str] = None
#       created_at: datetime = Field(default_factory=datetime.now)
#       updated_at: datetime = Field(default_factory=datetime.now)
#
#       def to_es_document(self) -> dict:
#           return self.model_dump(exclude={"id"})

File: Frontend/src/store/types/playbook.ts
// src/types/playbook.ts
// Types alignés avec le backend (SQLAlchemy) :
//   name, description, trigger, enabled, steps, variables,
//   timeout_seconds, max_retries, last_executed_at, execution_count, created_by

export type PlaybookTrigger = "manual" | "alert_created" | "alert_escalated" | "scheduled" | "webhook";

export interface PlaybookStep {
  action: "block_ip" | "disable_user" | "isolate_host" | "notify_slack" | "notify_email" | "create_ticket";
  params: Record<string, any>;
}

export interface Playbook {
  id: number;
  name: string;
  description: string | null;
  trigger: PlaybookTrigger;
  enabled: boolean;
  steps: PlaybookStep[];
  variables: Record<string, any>;
  timeout_seconds: number;
  max_retries: number;
  last_executed_at: string | null;
  execution_count: number;
  created_by: string | null;
  created_at: string;
  updated_at: string;
}